feat: use structs for queries

Satont opened the pull request:
Please let me know if you ok with this change, so i can continue with changes and cover all endpoints.

This approach will make it possible to type all the requests, and users will not have to constantly look in the documentation to see the available query parameters.

Satont committed:
feat: use structs for queries

mconnat reviewed (commented):
This is a better to handle the params than having an interface and then check the type sent.

If i understand well it will force us to create a params struct for every api endpoints ?
Thank you for your contribution.

Let's populate every calls and update the readme, then we merge it !

mconnat committed:
Players params as go struct

Remove params from request that don't need parameters in query.
Add go struct to handle params per face it endpoints.

mconnat committed:
test expanded params

Add params for expanded parameters needed as array of strings.
Update examples in Readme

mconnat committed:
Update README.md

Add missing from and to in the example

mconnat commented:
Can you review the 3 new commits on your PR please ?

I wanted to adapt the Players endpoints following your logic, test the expanded parameter cases (in championships) and update the Readme.

Satont commented:
> If i understand well it will force us to create a params struct for every api endpoints ? Thank you for your contribution.

Yep, but that's adding cover of params with structs, so users of your lib will know which query endpoint has.
